src/types/check_config.rs

```diff
@@ -91,17 +91,31 @@ impl CheckConfig {
             .map(|c| first_slot + (interval_secs * c))
             .collect()
     }
+
+    pub fn should_run(&self, tick: Tick, current_region: String) -> bool {
+        // Determines if this check should run in the current region at this current tick
+
+        // If region configs aren't present, assume we should always run this check
+        if self.active_regions.as_ref().is_none() || self.region_schedule_mode.is_none() {
+            return true;
+        }
```

Comment:
I think there's probably a more rusty way to write this to get the value into a variable otherwise return false, since now later you're having to do `unwrap`, but that unwrap isn't type checked. if you removed this guard here the compiler would still happily compile and just panic if it was None

Reply:
```rs
if self.region_schedule_mode.is_none() {
  return true;
}

let active_regions = Some(self.active_regions.as_ref()) else {
  return true;
}

let running_region_idx = (tick.time().timestamp() / self.interval as i64)
    .checked_rem(active_regions.len() as i64)
    .unwrap();

self.active_regions.as_ref().unwrap()[running_region_idx as usize] == current_region
```

Reply:
https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html